### DISCUSSION TITLE
Question about about IR and FTP Tuya protocols.

### DISCUSSION THREAD

**Discussion by msillano** [codetheweb/tuyapi#502] [General]
I found some devices, well managed by tuyAPI, which also use different protocols. Examples
1) [WiFi_IP_Camera](https://github.com/msillano/tuyaDAEMON/blob/main/devices/WiFi_IP_Camera/device_WiFi_IP_Camera.pdf) uses FTP to access the SD card and recorded videos.
2) [WiFi IR Remote Control Hub](https://www.aliexpress.com/item/4000031408686.html) uses an unknown protocol to send/receive IR commands, triggered by MQTT commands.

_ADDED_ IR info on [tuya documentation](https://developer.tuya.com/en/docs/cloud/universal-infrared).

Has anyone experienced implemented these protocols in js / node-red? I'm also looking for the specs ... I think some resources in TuyAPI can help implement it.

Best regards
m.s.

**Comment by lelemm** [comment]
Were you able to understand the wifi ir hub?

**Comment by msillano** [comment]
On my IR device, using tuyapi, I found only some data pushed by dp 1, values "send_ir" and "study_key".
It looks like a modal command.
The IR data must use another protocol for me unknown.(HTTP  ?)
I use it via TUYA automation, triggered by tuyaDAEMON.
Tath is with some limits, but it works.
To reduce this limits I asked about data protocol.
Example: to set clima temperature to 21 or 22 C, i need 2 separate automation, because TUYA automation are not parametric.

Still asking info about the protocol used

Best regards
m.s.
